jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java

```diff
@@ -973,18 +983,48 @@ public void reportFailures(FailureContext context) {
             }
         }
 
-        private void performStorySteps(RunContext context, List<PerformableSteps> storySteps, Stage stage)
-                throws InterruptedException {
+        private void performStorySteps(RunContext context, List<PerformableSteps> storySteps, Stage stage,
+                Optional<UUIDExceptionWrapper> causeToSkip) throws InterruptedException {
             context.reporter().beforeStorySteps(stage);
             for (PerformableSteps steps : storySteps) {
-                steps.perform(context);
+                steps.perform(context, causeToSkip);
             }
             context.reporter().afterStorySteps(stage);
         }
 
-        private void performScenarios(RunContext context) throws InterruptedException {
+        private void performScenarios(RunContext context, Optional<UUIDExceptionWrapper> causeToSkip)
+                throws InterruptedException {
             for (PerformableScenario scenario : scenarios) {
-                scenario.perform(context);
+                scenario.perform(context, causeToSkip.isPresent() ? causeToSkip : causeToSkip(scenario));
+                putScenarioId(scenario);
+            }
+        }
+
+        private Optional<UUIDExceptionWrapper> causeToSkip(PerformableScenario scenario) {
+            Meta meta = scenario.getScenario().getMeta();
+            if (meta.hasProperty(DEPENDS_ON_ID)) {
+                String dependsOnId = meta.getProperty(DEPENDS_ON_ID);
+                Boolean performDependentScenario = successfulScenariosId.get(dependsOnId);
+                if (performDependentScenario == null) {
+                    return wrap(new DependsOnNotFoundException(SCENARIO_WITH_ID + dependsOnId
+                            + " was not performed"));
```

Comment:
"Scenario with @scenarioId " + dependsOnId +" upon which current one depends is not found"